### ISSUE TITLE
supervisor failed to start  python Django program with the process entered RUNNING state

### ISSUE THREAD

**Issue by brightdark** [Supervisor/supervisor#1432]
When using supervisor to start a python Django program, the process entering RUNNING state, but actually the process failed to start since the web service provided by the program is unavailable. 

 **Information in detail:**

- OS: Debian GNU/Linux 9 \n \l,Linux BM1684-180 4.9.38-bm1684-v6.3.0-00428-gcce7bea #2 SMP Mon Sep 28 06:14:07 CST 2020 aarch64 GNU/Linux
- Python: Python 3.5.3 (default, Sep 27 2018, 17:25:39) [GCC 6.3.0 20170516] on linux
- - Django: 2.1
- supervisor: (apt install supervisor) supervisor/oldstable,now 3.3.1-1+deb9u1 all [installed] (V4.2.2 is also the same.)

**config detail:**
;supervisor config file
[program:web_backend]
directory=/workspace/platform/web_backend
command=/usr/bin/python3 manage.py runserver 0.0.0.0:18093
autostart=true
autorestart=true
priority=50
stdout_logfile=/workspace/log/supervisor/supervisord_backend_out.log
stdout_logfile_maxbytes=10MB
stdout_logfilebackups=1
stdout_capture_maxbytes=10MB
stderr_logfile=/workspace/log/supervisor/supervisord_backend_err.log
stderr_logfile_maxbytes=10MB
stderr_logfile_backups=1
stderr_capture_maxbytes=10MB
environment = HOME="/root", USER="root"

**supervisorctl status:**

web_backend                      RUNNING   pid 2153, uptime 0:35:32

**ps -eaf |grep python:**

root      1346     1  0 10:57 ?        00:00:03 /usr/bin/python /usr/bin/supervisord -n -c /etc/supervisor/supervisord.conf
root      2153  1346  0 10:57 ?        00:00:01 /usr/bin/python3 manage.py runserver 0.0.0.0:18093
root      2626  2153  3 10:57 ?        00:01:25 /usr/bin/python3 manage.py runserver 0.0.0.0:18093

**lsof -i:18093` or `netstat -tunlp | grep:18093**

no results

Any idea about the possible reason and how to solve the issue? 
Thx!

**Comment by mnaberez** [comment]
> supervisorctl status:
> 
> web_backend RUNNING pid 2153, uptime 0:35:32

`supervisorctl` claims that `supervisord` started the `web_backend` subprocess, that the subprocess is currently running, and it has a PID of `2153`.

> ps -eaf |grep python:
> 
> root 1346 1 0 10:57 ? 00:00:03 /usr/bin/python /usr/bin/supervisord -n -c /etc/supervisor/supervisord.conf
> root 2153 1346 0 10:57 ? 00:00:01 /usr/bin/python3 manage.py runserver 0.0.0.0:18093

`ps` shows that there is indeed a PID `2153` running and its PPID is that of `supervisord`.  This confirms the output from `supervisorctl`.

Based on the above, `supervisord` appears to be behaving normally.  It says that the program is running and it is.  There is likely a problem with the specific program you are trying to run.  The `stdout_logfile` and `stderr_logfile` of that program may have clues.  The Supervisor developers do not have enough bandwidth to assist with troubleshooting individual programs or installations, sorry.

**Comment by brightdark** [comment]
@mnaberez 

You are right. 

However, finally, I found that the issue is due to wrong setting of Python Environment Variable PYTHONPATH.

I have used a Python module depending on a .so file (eg. cv2.so) which is not under the default PYTHONPATH.
If I run the .py file directly in shell, it will abort at once with error info “no module named cv2”
However, when started by supervisord,  the program seems running normally. But in fact it failed. I've no idea what it is doing and why it is blocked in running state.  Whatever, after I set PYTHONPATH in correct value in supervisord config files, the program enters the right running state.

Thanks very much for your patient reply and hope the info above would be helpful for guys who encounter the same issue.
